internal/app/frontend/frontend_service.go

```diff
@@ -45,32 +57,92 @@ func newFrontend(cfg config.View) (*frontendService, error) {
 	return fs, nil
 }
 
-// CreateTicket will create a new ticket, assign a Ticket id to it and put the
-// Ticket in state storage. It will then look through the 'properties' field
-// for the attributes defined as indices the matchmakaking config. If the
-// attributes exist and are valid integers, they will be indexed. Creating a
-// ticket adds the Ticket to the pool of Tickets considered for matchmaking.
+// CreateTicket will create a new ticket, assign an id to it and put it in state
+// storage. It will index the Ticket attributes based on the indexing configuration.
+// Indexing a Ticket adds the it to the pool of Tickets considered for matchmaking.
 func (s *frontendService) CreateTicket(ctx context.Context, req *pb.CreateTicketRequest) (*pb.CreateTicketResponse, error) {
-	return &pb.CreateTicketResponse{}, nil
+	// Generate a ticket id and create a Ticket in state storage
+	ticket := proto.Clone(req.Ticket).(*pb.Ticket)
+	if ticket == nil {
```

Comment:
ticket wont be nil if the succeeded

Reply:
The input can be nil right? I ve moved the check directly on the input now. Added the casting error check here so gocli should be happy.